### PR TITLE
Proto for local RBAC.

### PR DESCRIPTION
**Pull request by liminw**
This API is added for implementing Istio RBAC on Envoy.

Design doc is at:
https://docs.google.com/document/d/1XkN9ls-ZDISz2uXGN20RaIre63DkJJOJNUOzcKCxbWA

### CODE REVIEW

File: rbac/v1alpha1/rbac.proto [hunk 1]
```diff
@@ -41,8 +41,8 @@ syntax = "proto3";
 //       namespace: istio-system
 //     spec:
 //       subject:
-//         user: request.auth.principal | ""
-//         groups: request.auth.principal | ""
+//         user: source.user | ""
```
Comment (yangminzhu) [inline rbac/v1alpha1/rbac.proto]: Why do we need to change the user to a different attribute, Is this part of the local RBAC change? Do we also need to change the corresponding setup yaml file?

File: rbac/v1alpha1/local_rbac.proto [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+//    },
+//  ]
+message LocalRBAC {
+  // A list of ServicePolicy. One per service.
+  repeated ServicePolicy service_policies = 1;
+}
+
+// ServicePolicy specifies a service and the Istio RBAC policies that apply to the service.
+message ServicePolicy {
+  // The name of the service.
```
Comment (yangminzhu) [inline rbac/v1alpha1/local_rbac.proto]: Could you make it clear this would be the full qualified name?

File: rbac/v1alpha1/local_rbac.proto [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+//         rules:
+//         - services: [“products.default.svc.cluster.local”]
+//           paths: [“/products/*”, “*/reviews”]
+//           methods: [“GET”, “HEAD”]
+//       bindings:
+//       - subjects:
+//         - user: “alice@yahoo.com”
+//         roleRef:
+//           kind: ServiceRole
+//           name: “service-admin”
```
Comment (yangminzhu) [inline rbac/v1alpha1/local_rbac.proto]: This should be "product-viewer"?

File: rbac/v1alpha1/local_rbac.proto (hunk 2 above)
Comment (liminw) [reply rbac/v1alpha1/local_rbac.proto]: Updated comments.

File: rbac/v1alpha1/local_rbac.proto (hunk 3 above)
Comment (liminw) [reply rbac/v1alpha1/local_rbac.proto]: Good catch! fixed.

File: rbac/v1alpha1/rbac.proto (hunk 1 above)
Comment (liminw) [reply rbac/v1alpha1/rbac.proto]: This is not really related to local RBAC change. I just updated the example here to be consistent with our sample.

**Pull request closed without merging**